signalfx/resource_signalfx_team_links.go

```diff
@@ -0,0 +1,183 @@
+package signalfx
+
+import (
+	"context"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func teamLinksResource() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"team": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Team ID to link the configured dashboard groups or detectos to.",
+			},
+			"dashboard_groups": {
+				Type:        schema.TypeSet,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: "Dashboard group IDs to link to the team.",
+				Optional:    true,
+			},
+			"detectors": {
+				Type:        schema.TypeSet,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: "Detector IDs to link to the team.",
+				Optional:    true,
+			},
+		},
+		Create: teamLinksCreate,
+		Read:   teamLinksRead,
+		Update: teamLinksUpdate,
+		Delete: teamLinksDelete,
+		Exists: teamLinksExists,
+	}
+}
+
+func teamLinksExists(d *schema.ResourceData, meta interface{}) (bool, error) {
+	return true, nil
+}
+
+func teamLinksRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*signalfxConfig)
+	teamId := d.Get("team").(string)
+	res, err := config.Client.GetTeam(context.TODO(), teamId)
+	if err != nil {
+		if strings.Contains(err.Error(), "404") {
```

Comment:
Setting id to blank string only on 404s is not entirely clear to me.  Can you explain the reasoning behind this in a comment?

Reply:
Since it's a virtual resource I don't think this case really matters. We perform the same logic whether it's an update or create (calling SetId("") removes the state resulting in a create on next tf run)